### PR TITLE
feat: Add New Redesign Of Issue Explorer (pushed to main)

### LINKED ISSUE #507: [Advanced]: Add Issue Explorer Banner Image
### 🧑‍🔬 Advanced Issue

Welcome! This is an **[Advanced Issue](https://github.com/issues?q=is%3Aopen%20is%3Aissue%20org%3Ahiero-ledger%20archived%3Afalse%20no%3Aassignee%20label%3A%22advanced%22%20repo%3Ahiero-ledger%2Fhiero-website)** touching core website architecture.

It is designed for expert contributors who have demonstrated deep architectural understanding and a proven track record of high-quality contributions.

### 🐞 Problem Description

Currently there is a missing banner image for the Issue Explore Page. 

We would like to add one from our new redesign. 

### 🛠️ Implementation Notes

Implement the banner image defined in our figma file specification

https://www.figma.com/design/YndiSK4Z4P2yXPxlsj3UHU/Hiero-Website-Design-Darshit?node-id=0-1&p=f&t=CYNOHXDCWnteDqt6-0

PUSH WORK TO websiteRedesign BRANCH

Markdown Specification: 
https://github.com/hiero-ledger/hiero-website/blob/main/docs/design-specs.md

**Blog & Content Workflow:**
- [Writing Blog Posts](https://github.com/hiero-ledger/hiero-website/blob/main/docs/blogs.md) — covers the current posting flow, front matter format, and content conventions

**Project Conventions:**
- Components live in `src/components/` — follow existing patterns for props, naming, and file structure
- Pages are under `src/app/` using the Next.js App Router
- Content (blog posts, etc.) lives in `content/`
- Styling uses [Tailwind CSS](https://tailwindcss.com/docs)

### 🧠 Advanced Contributors — Prerequisites & Expectations

> [!CAUTION]
> **Advanced issues are the highest-risk work in this project. We will reject PRs that do not meet these standards.**

### 🏁 Concrete Prerequisites
- **Advanced Web Development:** Proficient with React, Next.js, and TypeScript.
- **Expertise:** Deep architectural understanding of the website's component structure, routing, and build pipeline.
- **Proven History:** Successfully completed **≥ 10 intermediate issues** in this repo.
- **Consistency:** **≥ 3–4 months** of active, human-led contributions to this project.

> [!NOTE]
> **CI/CD Exception:** For issues focused on **GitHub Actions / Workflows**, the repo-specific thresholds above may be waived if the contributor demonstrates advanced-level proficiency in CI/CD.

### ⚠️ AI Usage Policy

- Using AI to generate code for Advanced issues is **strictly discouraged**
- AI may be used to help explain file relationships, but cannot be the main source of research.
- Submitting AI-generated or unvalidated code is grounds for **immediate closure**

### ⏱️ Timeline & Workflow
- **Typical time:** ~1 month / ~50 hours.
- 🔴 Completing an advanced issue in 1–3 days is a **red flag** and will likely be rejected.
- **Suggested:** Post your proposed architectural approach as a comment and wait for explicit maintainer approval **before writing any code.**

### 🔬 Technical Domains

- [ ] **Advanced React / Next.js** (Server Components, App Router, dynamic routing, middleware)
- [ ] **Component Architecture** (design systems, reusable patterns, composition)
- [ ] **Performance & Optimization** (Core Web Vitals, image optimization, code splitting, caching)
- [ ] **Accessibility** (WCAG compliance, screen reader support, keyboard navigation)
- [ ] **Build & Deployment** (Next.js config, CI/CD, hosting/deployment integration)
- [ ] **Testing** (unit, integration, E2E, accessibility testing, visual regression)

### 🛡️ Quality & Review Standards

Advanced PRs must be **"safe, maintainable, architecturally sound, and production-ready."**

1. **Architectural Fit:** The solution must fit naturally into the existing website architecture.
2. **Security & Correctness:** Evaluate all logic for injection risks, XSS vulnerabilities, or state corruption issues. 
3. **Maintainability:** Code must be short and clear enough for any other maintainer to debug without your assistance. 
4. **Backward Compatibility:** Existing pages, routes, and components must not break. If a breaking change is required, it must be explicitly managed.
5. **Comprehensive Testing:** Must include unit and integration tests covering all new logic paths, edge cases, and failure modes. AI-generated tests based on AI-generated code are grounds for immediate rejection.

### ✅ PR Quality Checklist

Before opening your PR, the contributor must confirm:
- [ ] I have spent the majority of my time researching the problem and solution space extensively, including reviewing relevant code, documentation, and external resources.
- [ ] I understand the system-wide impact of these changes on affected modules and performance.
- [ ] The design fits with the current website's architectural approaches.
- [ ] I have tested my changes extensively across browsers and screen sizes.
- [ ] Every line of code is personally understood and explainable.

### 📚 Resources & Support

**References:**
- [Next.js Documentation](https://nextjs.org/docs)
- [React Documentation](https://react.dev)
- [Tailwind CSS Documentation](https://tailwindcss.com/docs)
- [Vitest Documentation](https://vitest.dev)
- [Writing Blog Posts](https://github.com/hiero-ledger/hiero-website/blob/main/docs/blogs.md)

**🆘 Stuck?**
- [Community Calls](https://zoom-lfx.platform.linuxfoundation.org/meetings/hiero?view=week)
- [Discord](https://github.com/hiero-ledger/sdk-collaboration-hub/blob/main/guides/issue-progression/for-developers/discord.md)

### LINKED ISSUE #508: [Advanced]: Update Issue Explore Layout
### 🧑‍🔬 Advanced Issue

Welcome! This is an **[Advanced Issue](https://github.com/issues?q=is%3Aopen%20is%3Aissue%20org%3Ahiero-ledger%20archived%3Afalse%20no%3Aassignee%20label%3A%22advanced%22%20repo%3Ahiero-ledger%2Fhiero-website)** touching core website architecture.

It is designed for expert contributors who have demonstrated deep architectural understanding and a proven track record of high-quality contributions.

### 🐞 Problem Description

Currently the issue explorer layout is missing an eye catching UI

We would like to expand the design follow our outlined specification in figma

### 🛠️ Implementation Notes

Implement the design according to the figma file found here 

https://www.figma.com/design/YndiSK4Z4P2yXPxlsj3UHU/Hiero-Website-Design-Darshit?node-id=0-1&p=f&t=CYNOHXDCWnteDqt6-0

The design should include features that enhance a users accessibility of our site. 

The desgin should follow the specification outlined in our figma file.

Note:
For the skill labels, our team would like these positioned next to the repository labels. 

An example of this can be found here: https://www.figma.com/design/hZztcL7ukzpYLlIklBFzwP/Hiero-Website-Design-Daniel?node-id=0-1&p=f&t=3UYJ7D1bhd7NWafv-0

PUSH WORK TO websiteRedesign BRANCH

Markdown Specification: 
https://github.com/hiero-ledger/hiero-website/blob/main/docs/design-specs.md

**Blog & Content Workflow:**
- [Writing Blog Posts](https://github.com/hiero-ledger/hiero-website/blob/main/docs/blogs.md) — covers the current posting flow, front matter format, and content conventions

**Project Conventions:**
- Components live in `src/components/` — follow existing patterns for props, naming, and file structure
- Pages are under `src/app/` using the Next.js App Router
- Content (blog posts, etc.) lives in `content/`
- Styling uses [Tailwind CSS](https://tailwindcss.com/docs)

### 🧠 Advanced Contributors — Prerequisites & Expectations

> [!CAUTION]
> **Advanced issues are the highest-risk work in this project. We will reject PRs that do not meet these standards.**

### 🏁 Concrete Prerequisites
- **Advanced Web Development:** Proficient with React, Next.js, and TypeScript.
- **Expertise:** Deep architectural understanding of the website's component structure, routing, and build pipeline.
- **Proven History:** Successfully completed **≥ 10 intermediate issues** in this repo.
- **Consistency:** **≥ 3–4 months** of active, human-led contributions to this project.

> [!NOTE]
> **CI/CD Exception:** For issues focused on **GitHub Actions / Workflows**, the repo-specific thresholds above may be waived if the contributor demonstrates advanced-level proficiency in CI/CD.

### ⚠️ AI Usage Policy

- Using AI to generate code for Advanced issues is **strictly discouraged**
- AI may be used to help explain file relationships, but cannot be the main source of research.
- Submitting AI-generated or unvalidated code is grounds for **immediate closure**

### ⏱️ Timeline & Workflow
- **Typical time:** ~1 month / ~50 hours.
- 🔴 Completing an advanced issue in 1–3 days is a **red flag** and will likely be rejected.
- **Suggested:** Post your proposed architectural approach as a comment and wait for explicit maintainer approval **before writing any code.**

### 🔬 Technical Domains

- [ ] **Advanced React / Next.js** (Server Components, App Router, dynamic routing, middleware)
- [ ] **Component Architecture** (design systems, reusable patterns, composition)
- [ ] **Performance & Optimization** (Core Web Vitals, image optimization, code splitting, caching)
- [ ] **Accessibility** (WCAG compliance, screen reader support, keyboard navigation)
- [ ] **Build & Deployment** (Next.js config, CI/CD, hosting/deployment integration)
- [ ] **Testing** (unit, integration, E2E, accessibility testing, visual regression)

### 🛡️ Quality & Review Standards

Advanced PRs must be **"safe, maintainable, architecturally sound, and production-ready."**

1. **Architectural Fit:** The solution must fit naturally into the existing website architecture.
2. **Security & Correctness:** Evaluate all logic for injection risks, XSS vulnerabilities, or state corruption issues. 
3. **Maintainability:** Code must be short and clear enough for any other maintainer to debug without your assistance. 
4. **Backward Compatibility:** Existing pages, routes, and components must not break. If a breaking change is required, it must be explicitly managed.
5. **Comprehensive Testing:** Must include unit and integration tests covering all new logic paths, edge cases, and failure modes. AI-generated tests based on AI-generated code are grounds for immediate rejection.

### ✅ PR Quality Checklist

Before opening your PR, the contributor must confirm:
- [ ] I have spent the majority of my time researching the problem and solution space extensively, including reviewing relevant code, documentation, and external resources.
- [ ] I understand the system-wide impact of these changes on affected modules and performance.
- [ ] The design fits with the current website's architectural approaches.
- [ ] I have tested my changes extensively across browsers and screen sizes.
- [ ] Every line of code is personally understood and explainable.

### 📚 Resources & Support

**References:**
- [Next.js Documentation](https://nextjs.org/docs)
- [React Documentation](https://react.dev)
- [Tailwind CSS Documentation](https://tailwindcss.com/docs)
- [Vitest Documentation](https://vitest.dev)
- [Writing Blog Posts](https://github.com/hiero-ledger/hiero-website/blob/main/docs/blogs.md)

**🆘 Stuck?**
- [Community Calls](https://zoom-lfx.platform.linuxfoundation.org/meetings/hiero?view=week)
- [Discord](https://github.com/hiero-ledger/sdk-collaboration-hub/blob/main/guides/issue-progression/for-developers/discord.md)

### PR DESCRIPTION
**Pull request by aceppaluni**
# Description

## Summary

Redesign the Issue Explorer page to better align with the Hiero website's visual design and design specifications.

## Changes

- Redesigned the Issue Explorer hero section with:
  - Hiero brand gradient background
  - Large "Issue Explorer" heading
  - Open-source / Hiero SDKs label
  - Open issue count and contribution badges
  - Decorative network-style background elements
- Redesigned the filter bar with:
  - Styled difficulty selector
  - Styled repository selector
  - Issue count indicator
- Redesigned issue results as responsive cards:
  - Repository badge
  - Issue title
  - Difficulty badge
  - Hover states and issue navigation arrow
- Added responsive layouts:
  - Single-column layout on mobile
  - Two-column layout on tablet
  - Four-column layout on large screens
- Applied the Hiero design specification colors and typography.
- Added visual styling for the four difficulty levels:
  - Good First Issue
  - Beginner
  - Intermediate
  - Advanced
- Preserved the existing issue filtering and GitHub issue-link behavior.

## Design

The updated layout is based on the provided Issue Explorer designs and follows the existing Hiero design system, including:

- Space Grotesk typography
- Hiero red palette
- Charcoal and neutral colors
- Existing type scale conventions
- Consistent borders, spacing, and card styling

## Testing

- Verified the Issue Explorer renders correctly with the updated layout.
- Verified difficulty and repository filters continue to function.
- Verified issue cards link to the corresponding GitHub issues.
- Verified the layout responds across different screen sizes.

## Related Issues

Fixes: #507
Fixes: #508 

## Checklist
- [ ] Tests added/updated
- [ ] Documentation updated
- [ ] Linting passes
- [x] Branch up-to-date with main

## Notes
This has been pushed to the main branch for now so we may view this in netlify. 

I will push this to the proper branch once approvals have been added. 

Please also note: that labels such as "Beginner, Good First Issue" may appear on one issue. This is because some SDK's for example C++ have added both labels to their issues. I have tested this by removing a label and it does correct this. I have left the other issues this way as I did not make the issues for other respective repositories. 

## Summary by CodeRabbit

- **New Features**
  - Redesigned the Issue Explorer with a more engaging hero section.
  - Added styled filter controls for easier issue browsing.
  - Introduced responsive issue cards for improved viewing across screen sizes.
  - Added difficulty and repository badges to provide clearer issue context.

### CODE REVIEW

**Review by exploreriii — COMMENTED** [review]
What are your thoughts on default sorting the rendered issues by difficulty?

i.e. GFI -> beginner -> intermediate -> advanced -> null

This makes it easier to quickly locate what sort of issues you are looking for


We could either default sort by difficulty for a given REPO i.e. show all python ones first, then javascript 

or, we can think users will not mind which repo, and just sort by difficulty for all the repos

Alternatively, we can consider grouping the return by repo type
i.e. SDKs / Core / Documentation

Secondly I would argue we can change this: Find open issues across Hiero SDKs and start contributing
to:
Find open issues across Hiero and start contributing
(assuming some of the issues will belong to the website or other repos)

**Comment by exploreriii** [comment]
Another idea could be to support multiple organisations. i.e. the user can get a returned list of issues at hiero ledger, but could also filter to select issues at hiero hackers if that is desirable

**Comment by aceppaluni** [comment]
I like your ideas! 

For you first idea, I would recommend opening an issue for this as I had wanted to keep the scope of this to the redesign. Plus opening an issue/s will give us more for contributors :)

I did update the find text :)

Thank you!!

File: src/app/issues/page.tsx [hunk 1]
```diff
@@ -21,63 +86,216 @@ export default function IssueExplorer() {
...
+                <span className="rounded-full border border-white/20 bg-white/15 px-5 py-2.5 text-sm font-medium text-white backdrop-blur-sm">
+                  Ready to contribute
+                </span>
+              </div>
+            </div>
+          </div>
+        </Container>
+      </section>
+
+      {/* Filters */}
```
Comment (exploreriii) [inline src/app/issues/page.tsx:538]: This is hard to read as there are a lot of components with custom functionality and css in the same function

can i suggest in this issue or follow ups migrating to:

creating components:
- issueCard & test
- IssueExplorerHero & test
- IssueFilters & test

This would allow keeping issues/page.tsx streamlined to just the execution of these called components, more similar to e.g. heros page and hacktoberfest page

Comment (exploreriii) [reply src/app/issues/page.tsx:538]: similarly, for the logic side of the page (like issue fetching, filtering and types, we can split into maybe src/issues/difficulty.ts, src/issues/filter.ts, etc. 

This way we can keep the logic on how difficulty is established, how issues are filtered etc in reviewable sections, with individual tests

File: src/app/issues/page.tsx [hunk 2]
```diff
@@ -21,63 +86,216 @@ export default function IssueExplorer() {
...
+
+              <div className="my-8 h-px w-24 bg-white" />
+
+              <p className="max-w-[520px] text-lg leading-7 tracking-[-0.02em] text-white/80 sm:text-xl">
+                Find open issues across Hiero and start contributing!
+              </p>
+
+              <div className="mt-8 flex flex-wrap gap-3">
+                <span className="rounded-full border border-white/20 bg-white/15 px-5 py-2.5 text-sm font-medium text-white backdrop-blur-sm">
+                  {issues.length} open issues
```
Comment (exploreriii) [inline src/app/issues/page.tsx:526]: this is a count of issues length, but it is hard to see if this is correctly paginating, so the count might not be accurate --  by separating out the fetch, we can see more clearly if it is accurate (and will continue to be so, as repos introduce more issues, or, if we want to explicitly cap)

File: src/app/issues/page.tsx [hunk 3]
```diff
@@ -6,6 +6,71 @@ import { useState } from "react";
 import { useIssues } from "@/hooks/useIssues";
 import { GitHubIssue } from "@/issues/types";
 
+type Difficulty = {
+  label: string;
+  value: string;
+};
+
+const difficultyAliases: Record<string, string[]> = {
```
Comment (exploreriii) [inline src/app/issues/page.tsx:14]: this can introduce drift because there is also src/issues/filter.ts, ideally we use the existing difficutly map

File: src/app/issues/page.tsx [hunk 4]
```diff
@@ -21,63 +86,216 @@ export default function IssueExplorer() {
...
+                  className="h-10 min-w-[140px] appearance-none rounded-lg border border-gray-light bg-white px-4 pr-10 text-sm font-medium text-charcoal outline-none transition-colors hover:border-gray focus:border-red focus:ring-2 focus:ring-red-light/20">
+                  <option value="">All Difficulties</option>
+                  <option value="good first issue">Good First Issue</option>
+                  <option value="beginner">Beginner</option>
+                  <option value="intermediate">Intermediate</option>
+                  <option value="advanced">Advanced</option>
+                </select>
+
+                <span className="pointer-events-none absolute right-3 top-1/2 -translate-y-1/2 text-xs text-gray">
+                  ▾
```
Comment (exploreriii) [inline src/app/issues/page.tsx:557]: should these also have aria hidden?

File: src/app/issues/page.tsx [hunk 5]
```diff
@@ -21,63 +86,216 @@ export default function IssueExplorer() {
   } = useIssues(difficulty, sdk);
 
   return (
-    <Container>
-      <div className="mb-10">
-        <h1 className="text-2xl mb-2.5 sm:text-4xl sm:mb-5">Issue Explorer</h1>
-        <RichText
-          className="text-lg max-w-full md:max-w-[800px]"
-          markdown="Browse open issues across the Hiero SDKs by difficulty. The four levels below are the rungs of the [Issue Progression Initiative](/blog/hiero-issue-progression-initiative)."
```
Comment (exploreriii) [inline src/app/issues/page.tsx:29]: is it still importantt o link to IPI blog?

File: src/app/issues/page.tsx (hunk 3 above)
Comment (exploreriii) [reply src/app/issues/page.tsx:14]: I think we currently have difficulty in a few places - this will become a problem, we should try to centralise and use the same definitions

File: src/app/issues/page.tsx [hunk 6]
```diff
@@ -6,6 +6,71 @@ import { useState } from "react";
 import { useIssues } from "@/hooks/useIssues";
```
Comment (exploreriii) [inline src/app/issues/page.tsx:6]: we can consider adding to globals.css things like focus ring, issue card, difficulty badge so we don't have to maintain inline copies

**Review by exploreriii — APPROVED** [review]
Hi @aceppaluni i think this content is OK to merge but maintainbility and accuracy concerns could be valid to tackle in future issues, if you are able/want to create

**Review by exploreriii — CHANGES_REQUESTED** [review]
Blocking as PR is incorectly made to main, should be the redesign branch

**Comment by darshit2308** [comment]
@aceppaluni  Hi, in the issue explorer section, I see that the background is completely static.
image

Don't we want a dynamic moving background ? Or, this is what the community decided ?

https://github.com/user-attachments/assets/29eb2511-442a-4fd6-b92f-fe9be071db82

**Comment by aceppaluni** [comment]
> Blocking as PR is incorectly made to main, should be the redesign branch

Yup, did this so we can see preview. Will make sure to push to proper branch :)

**Comment by aceppaluni** [comment]
> @aceppaluni  Hi, in the issue explorer section, I see that the background is completely static.
> 
> image
> 
> 
> 
> Don't we want a dynamic moving background ? Or, this is what the community decided ?
> 
> 
> 
> https://github.com/user-attachments/assets/29eb2511-442a-4fd6-b92f-fe9be071db82
> 
> 
> 
> 

Yes, however, We have a version 4 coming soon and am reaching out to the team to see how they want to progress. Thank you!! :)

**Comment by darshit2308** [comment]
@aceppaluni  Oh got it 👍, and if there is any particular help required, then I would be more then happy to help 😄

**Comment by aceppaluni** [comment]
@darshit2308 Thank you!!

I have gone ahead and added the updated background. I think this makes a much nicer addition but will confirm with the team! I hope you like it! :)

**Comment by aceppaluni** [comment]
This PR is intentionally pushed to main so we can see the preview from netlify. 

PR can remain in draft until approvals are final then I will push to the proper website redesign branch.

Thank you!

**Comment by darshit2308** [comment]
@aceppaluni  Yes, this one looks really good to me, thanks !!

**Comment by aceppaluni** [comment]
closing as PR has been pushed to proper branch
